File: money_src_junit/src/fr/ulille1/fil/odeva/Money.java
package  fr.ulille1.fil.odeva;


public class Money {
	private int value;
	private String currency;
	

	Money(int value, String currency)
	{
		this.value=value;
		this.currency=currency;
	}

	public int getValue()
	{
		return this.value;
	}

	public String getCurrency()
	{
		return this.currency;
	}
	
	public boolean _equals(Object m){
		if(m instanceof Money) {
			Money money = (Money)m;
			return this.getValue() == money.getValue() && this.getCurrency() == money.getCurrency();
		}
		else
			return false;
	}
	
	public boolean equals(Object m) {
		return _equals(m);
	}

    public String toString() {
     	return this.getValue()+" ("+this.getCurrency()+")";
    }

}

File: money_js_qunit/src/money.js
var money= (function(){
	function money(v,curr) {
		if(v < 0)
			throw new ValueNegativExc(v);
		this.v=v;
		if(curr.length != 3)
			throw new CurrencyLengthExc(curr);
		this.curr=curr;
	}

	money.prototype.getCurrency=function () {
		return this.curr;
	}
	money.prototype.getValue=function () {
		return this.v;
	}
	money.prototype.equals=function (otherM) {
		return (otherM.getValue()==this.getValue() && otherM.getCurrency().toLowerCase()==this.getCurrency().toLowerCase()) ;
	}
	money.prototype.toString=function toString() {
		return this.getValue()+" ("+this.getCurrency()+")" ;
	}


return money;
})();
